pkg/apis/helm.cattle.io/v1/types.go

```diff
@@ -32,6 +32,7 @@ type HelmChartSpec struct {
 	Bootstrap       bool                          `json:"bootstrap,omitempty"`
 	ChartContent    string                        `json:"chartContent,omitempty"`
 	JobImage        string                        `json:"jobImage,omitempty"`
+	BackOffLimit    int32                         `json:"backOffLimit,omitempty"`
```

Comment:
I think this would be better as an int pointer, otherwise the `omitempty` doesn't make sense as there is no such thing as empty/nil for a basic type. This should make the rest of the code flow better as the Job's BackoffLimit field is also `*int32`. Ref: https://github.com/kubernetes/api/blob/v0.27.4/batch/v1/types.go#L263
```suggestion
	BackOffLimit    *int32                        `json:"backOffLimit,omitempty"`
```

Reply:
sounds good.. i will make the change.